packages/experimental/Button/src/CompoundButton/CompoundButton.tsx

```diff
@@ -29,7 +30,20 @@ const CompoundButtonComposed = compose<CompoundButtonType>({
 
     // now return the handler for finishing render
     return (final: CompoundButtonPropsWithInnerRef, ...children: React.ReactNode[]) => {
-      const { icon, secondaryContent, iconPosition, accessibilityLabel, ...mergedProps } = mergeProps(button.props, final);
+      const { icon, iconOnly, secondaryContent, iconPosition, loading, accessibilityLabel, ...mergedProps } = mergeProps(
+        button.props,
+        final,
+      );
+
+      const shouldShowIcon = !loading && icon;
```

Comment:
We'll need to find a way how to avoid code duplication. Part with markup is similar in all Button components. Maybe we can move it in one file and reuse. It's just an idea, not for this PR